packages/protocol/contracts/layer1/core/impl/Inbox.sol

```diff
@@ -464,6 +473,30 @@ contract Inbox is IInbox, ICodec, IForcedInclusionStore, EssentialContract {
         return _proposalHashes[_proposalId % _ringBufferSize];
     }
 
+    // ---------------------------------------------------------------
+    // Bond Manager Functions
+    // ---------------------------------------------------------------
+
+    /// @inheritdoc IBondManager
+    function deposit(uint256 _amount) external nonReentrant {
```

Comment:
A potentially cleaner alternative is to introduce a BondManagerBase.sol that implements IBondManager, and then have Inbox inherit from BondManagerBase.

That said, I still think adopting ProverAuction directly is the better approach overall. It would also let us deprecate IProverWhitelist.

The main selling point of ProverAuction is that it completely hides bond accounting from Inbox.